File: 334-increasing-triplet-subsequence/solution.py
import sys

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        size = len(nums)
        mins = [sys.maxsize] * 3

        for n in nums:
            if n < mins[0]:
                mins[0] = n
            if n > mins[0]:
                mins[1] = min(n, mins[1])
            if n > mins[1]:
                return True
        
        return False
        

File: 334-increasing-triplet-subsequence/solution.go
package main

import (
	"math"
)

func increasingTriplet(nums []int) bool {
	mins := make([]int, 3)
	for i := range mins {
		mins[i] = math.MaxInt
	}

	for _, val := range nums {
		if val < mins[0] {
			mins[0] = val
		}
		if val > mins[0] {
			mins[1] = int(math.Min(float64(val), float64(mins[1])))
		}
		if val > mins[1] {
			return true
		}
	}

	return false
}
